contracts/separate_value_bridge.md

```diff
@@ -0,0 +1,103 @@
+# Proposal for Message Bus native ETH handling
+
+In the current implementation we have a Message Bus (MB) that acts as:
+
+1. A transport layer between Ten and Ethereum of generic messages of the form:
+```solidity
+    event LogMessagePublished 
+    (
+        address sender, 
+        uint64 sequence, 
+        uint32 nonce, 
+        uint32 topic, 
+        bytes payload, 
+        uint8 consistencyLevel,
+    );
+```
+
+where the `payload` is the message arbitrarily encoded by the top-of-stack bridge.
+For example an "ERC20" transfer command to be executed by the ERC20 contract.
+
+2. A "native ETH bridge" between Ethereum and TEN.
+
+The MB owns native ETH on the Ethereum side and will dispense it to the receiver when receiving a message like this:
+```solidity
+    event ValueTransfer
+    (
+        address indexed sender,
+        address indexed receiver,
+        uint256 amount,
+        uint64 sequence
+    );
+```
+On TEN, the MB does not own ETH, but the MB is a special "System contract", and the platform itself mints and burns ETH 
+on messages such as this. 
+
+
+```plantuml
+@startuml
+'https://plantuml.com/sequence-diagram
+
+title Current Ten->Ethereum native value withdrawal flow
+autonumber
+
+Alice -> TEN: Calls "EthereumBridge.sendNative" with 0.1Eth
+TEN --> TEN: Emit MB `ValueTransfer` event encoding the VT (E1)
+
+TEN -> Ethereum: Rollup with Xchain MTreeRoot (R1)
+
+Alice -> Ethereum: Call "CrossChain.extractNativeValue" with E1 as parameter. (This is the "Relay" step)
+Ethereum -> Ethereum: Check E1 against R1 (Call the Message Bus contract)
+Ethereum -> Ethereum: Move native value from the MB to Alice's account (`receiver.call{value: amount}("");`)
+note bottom
+Alice - can be another smart contract
+end note
+@enduml
+```
+
+## Requirements
+
+1. Deposit and Withdraw native ETH into EOA.
+
+2. Deposit and Withdraw native ETH into smart contracts by calling custom functions.
+   - This is required by liquidity bridges. (Stefan - proof?)
```

Comment:
I gotta check the contracts they got, but the basic premise is they need to move the native eth they give out around chains and the only way to move it is through the native bridge; as i said those bridges are DAOs so they can't be sending to private keys and need to be able to call contracts and pay them to execute the settling and repay the bonder/relayer who has done the early transfer to the user